### PR TITLE
Started working on audio, only sending audio work at the moment

### PR DESCRIPTION
**Pull request by lclc98**
It current has support for WAV, OGG, FLAC and MP3.
I will be working on javadocs while this gets reviewed.

### CODE REVIEW

File: src/main/java/sx/blah/discord/handle/impl/events/UserVoiceStateUpdateEvent.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import sx.blah.discord.handle.obj.IUser;
+
+public class UserVoiceStateUpdateEvent extends Event {
+    private final IUser user;
+    private final IGuild guild;
+
+    public UserVoiceStateUpdateEvent(IUser user, IGuild guild) {
+        this.user = user;
+        this.guild = guild;
+    }
```
Comment (austinv11) [inline src/main/java/sx/blah/discord/handle/impl/events/UserVoiceStateUpdateEvent.java]: Can you add getters for the user and guild object?

File: src/main/java/sx/blah/discord/handle/impl/obj/VoiceChannel.java [hunk 2]
```diff
@@ -14,90 +18,100 @@
...
-	
+
 	public VoiceChannel(IDiscordClient client, String name, String id, IGuild parent, String topic, int position, List<IMessage> messages, Map<String, PermissionOverride> roleOverrides, Map<String, PermissionOverride> userOverrides) {
 		super(client, name, id, parent, topic, position, messages, roleOverrides, userOverrides);
 	}
-	
+
+	@Override
+	public void joinChannel()
+	{
```
Comment (austinv11) [inline src/main/java/sx/blah/discord/handle/impl/obj/VoiceChannel.java]: Minor thing, but I would prefer that this bracket didn't have its own line.

**Comment by austinv11** [comment]
Looks good, just fix the things I commented on and add javadocs for the json response/requests and events and other added things and it should be good to go.

**Comment by lclc98** [comment]
I am going to remake this PR as it will be too hard to fix.

**Pull request closed without merging**